lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -318,31 +318,47 @@ export class PublicClientApplication {
     // #endregion
 
     // #region Silent Flow
-
+    
+    /**
+     * This function uses a hidden iframe to fetch an authorization code from the eSTS. There are cases where this may not work:
+     * - Any browser using a form of Intelligent Tracking Prevention
+     * - If there is not an established session with the service
+     * 
+     * In these cases, the request must be done inside a popup or full frame redirect. 
+     * 
+     * For the cases where interaction is required, you cannot send a request with prompt=none.
+     * 
+     * If your refresh token has expired, you can use this function to fetch a new set of tokens silently as long as 
+     * you session on the server still exists.
+     * @param {@link AuthenticationParameters} 
+     * 
+     * To renew idToken, please pass clientId as the only scope in the Authentication Parameters.
+     * @returns {Promise.<TokenResponse>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
+     */
     async ssoSilent(request: AuthenticationParameters): Promise<TokenResponse> {
         // block the reload if it occurred inside a hidden iframe
         BrowserUtils.blockReloadInHiddenIframes();
 
+        // Check that we have some SSO data
+        if (StringUtils.isEmpty(request.loginHint) && StringUtils.isEmpty(request.sid) && !request.account) {
+            throw BrowserAuthError.createSilentSSOInsufficientInfoError();
+        }
+
+        // Check that prompt is set to none, log message if it is set to anything else.
+        if (request.prompt && request.prompt !== PromptValue.NONE) {
+            console.log(`Prompt must be set to none for silent requests. The given value was ${request.prompt}`);
```

Comment:
I think this should be output to `logger.warn` instead of `console.log`.

Reply:
There is no logger object in the browser library currently, leaning on system console for now. I need to get Logger added soon.

Reply:
Should this resolve as an error?

Reply:
should this be an error?

Reply:
Initially I wanted to replace it for them, since calling ssoSilent would imply that they want a prompt=none. I think it may be better to throw an error here however. I have updated the code.